cmd/compile.go

```diff
@@ -0,0 +1,84 @@
+package cmd
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+
+	salv1 "github.com/speechly/api/go/speechly/sal/v1"
+)
+
+var compileCmd = &cobra.Command{
+	Use: "compile [directory]",
+	Example: `speechly compile -a UUID_APP_ID .
+speechly compile -a UUID_APP_ID /usr/local/project/app`,
+	Short: "Compiles a sample of examples from the given configuration",
+	Long: `The contents of the directory given as argument is sent to the
+API and compiled. If suffcessful, a sample of examples are printed to stdout.`,
+	Args: cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := cmd.Context()
+		appId, _ := cmd.Flags().GetString("app")
+		inDir := args[0]
+		absPath, _ := filepath.Abs(inDir)
+		log.Printf("Project dir: %s\n", absPath)
+		// create a tar package from files in memory
+		uploadData := createTarFromDir(inDir)
```

Comment:
The intermediate `tar` file validation is identical to `validate`, this should be shared code.